src/EditorFeatures/Test/SolutionCrawler/WorkCoordinatorTests.cs

```diff
@@ -1214,6 +1252,12 @@ public Task AnalyzeDocumentAsync(Document document, SyntaxNode bodyOpt, Invocati
             #endregion
         }
 
+        [ExportWorkspaceService(typeof(IGlobalOperationNotificationService), SolutionCrawler), Shared]
+        private class TestGlobalOperationService : GlobalOperationNotificationService
```

Comment:
❓ Why not move `GlobalOperationNotificationServiceFactory` to Workspaces where the interface is defined?

❗️ If this is included, it needs `[PartNotDiscoverable]` as well (attribute indicates that it's only included via an explicit `WithPart`).

Reply:
> you mean don't have default service but all uses ServiceLayer.Host? feels like we are removing all nice layer separation we used to have..

The Workspaces project currently includes an implementation of `IGlobalOperationNotificationService` which fails to adhere to the pre- and post-conditions of the methods it contains. The implementation in ServicesVisualStudio has no dependencies outside of Workspaces; moving it down to be the new default means two things:

1. The default implementation is no longer broken
2. The implementation in test code is redundant, per (1)

Reply:
I submitted a separate pull request #25960 to simplify this.